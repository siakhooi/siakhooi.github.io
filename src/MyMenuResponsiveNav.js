import React from 'react'
import PropTypes from 'prop-types'
import MyMenuTitle from './MyMenuTitle'
import MyMenuBars from './MyMenuBars'

MyMenuResponsiveNav.propTypes = {
  onClick: PropTypes.func,
  title: PropTypes.string
}

function MyMenuResponsiveNav (props) {
  return (
    <div className="MyMenuResponsiveNav">
      <MyMenuTitle title={props.title} />
      <MyMenuBars onClick={props.onClick} />
    </div>
  )
}

export default MyMenuResponsiveNav
