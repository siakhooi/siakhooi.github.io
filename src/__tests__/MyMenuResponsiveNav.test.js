import React from 'react'
import renderer from 'react-test-renderer'
import MyMenuResponsiveNav from '../MyMenuResponsiveNav'

it('MyMenuResponsiveNav/1', () => {
  const dummyFunction = jest.fn()

  const component =
    renderer.create(
      <MyMenuResponsiveNav title='Hello World' onClick={dummyFunction} />
    )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree
    .children.find((x) => { return x.props.className.indexOf('MyMenuBars') !== -1 })
    .props.onClick()
  expect(dummyFunction).toHaveBeenCalled()
})
