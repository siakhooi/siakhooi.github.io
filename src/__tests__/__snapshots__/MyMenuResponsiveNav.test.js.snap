// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyMenuResponsiveNav/1 1`] = `
<div
  className="MyMenuResponsiveNav"
>
  <span
    className="MyMenuTitle"
  >
    Hello World
  </span>
  <button
    className="MyMenuBars fa fa-bars"
    onClick={[Function]}
  />
</div>
`;
